checkov/common/graph/graph_builder/graph_components/blocks.py

```diff
@@ -93,54 +95,34 @@ def get_hash(self) -> str:
         if len(attribute_key_parts) == 1:
             self.attributes[attribute_key] = attribute_value
-            self.changed_attributes[attribute_key] = previous_breadcrumbs
+            if self._should_set_changed_attributes(change_origin_id, attribute_at_dest):
+                self.changed_attributes[attribute_key] = previous_breadcrumbs
             return
         for i in range(len(attribute_key_parts)):
             key = join_trimmed_strings(char_to_join=".", str_lst=attribute_key_parts, num_to_trim=i)
             if key.find(".") > -1:
                 self.attributes[key] = attribute_value
                 attribute_value = {attribute_key_parts[len(attribute_key_parts) - 1 - i]: attribute_value}
-                self.changed_attributes[key] = previous_breadcrumbs
+                if self._should_set_changed_attributes(change_origin_id, attribute_at_dest):
+                    self.changed_attributes[key] = previous_breadcrumbs
 
-    def update_inner_attribute(
-        self, attribute_key: str, nested_attributes: Union[List[Any], Dict[str, Any]], value_to_update: Any
-    ) -> None:
-        split_key = attribute_key.split(".")
-        i = 1
-        curr_key = ".".join(split_key[0:i])
-        if isinstance(nested_attributes, list):
-            if curr_key.isnumeric():
-                curr_key_int = int(curr_key)
-                if curr_key_int < len(nested_attributes):
-                    if not isinstance(nested_attributes[curr_key_int], dict):
-                        nested_attributes[curr_key_int] = value_to_update
-                    else:
-                        self.update_inner_attribute(
-                            ".".join(split_key[i:]), nested_attributes[curr_key_int], value_to_update
-                        )
-            else:
-                for inner in nested_attributes:
-                    self.update_inner_attribute(curr_key, inner, value_to_update)
-        elif isinstance(nested_attributes, dict):
-            while curr_key not in nested_attributes and i <= len(split_key):
-                i += 1
-                curr_key = ".".join(split_key[0:i])
-            if attribute_key in nested_attributes.keys():
-                nested_attributes[attribute_key] = value_to_update
-            if len(split_key) == 1 and len(curr_key) > 0:
-                nested_attributes[curr_key] = value_to_update
-            elif curr_key in nested_attributes.keys():
-                self.update_inner_attribute(".".join(split_key[i:]), nested_attributes[curr_key], value_to_update)
+    @staticmethod
+    def _should_add_previous_breadcrumbs(change_origin_id: Optional[int],
+            previous_breadcrumbs: List[BreadcrumbMetadata], attribute_at_dest: Optional[str]):
+        return not previous_breadcrumbs or previous_breadcrumbs[-1].vertex_id != change_origin_id
+
+    @staticmethod
+    def _should_set_changed_attributes(change_origin_id: Optional[int], attribute_at_dest: Optional[str]):
+        return True
```

Comment:
is this pre-work for a future update?

Reply:
`CloudformationBlocks` class overrides this method but `TerraformBlocks` uses this method as is